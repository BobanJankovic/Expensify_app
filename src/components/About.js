import React from 'react';
export const About = () => (
    <div>
    This is from  About
    </div>
);

 