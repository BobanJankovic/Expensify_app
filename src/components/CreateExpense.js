import React from 'react';
import { connect } from 'react-redux';
import AddForm from '../components/AddExpense';
import { addExpense } from '../actions/expenses';


const CreateExpense = (props) => (
    <div>
    <AddForm
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense));
        props.history.push('/');
        
      }}
    />
    </div>
);

 const mapStateToProps = (state)=> {
  return {
    expenses:state.expenses,
    filters:state.filters
  };
};
export default connect()(CreateExpense);