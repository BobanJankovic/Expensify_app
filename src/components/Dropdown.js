import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import {sortByAmount} from '../actions/filters';
import {sortByDate} from '../actions/filters';

const Dropdown= (props)=> (
  <div>
    <select value={props.filters.sortBy} 
      onChange={
                (e)=>{
                  if (e.target.value==="date") {
                    props.dispatch(sortByDate())
                  } else if ((e.target.value==="amount")){
                    props.dispatch(sortByAmount())
                  }
                }
              }
      >
      <option value="amount" >Amount</option>
      <option value="date" >Date</option>
      
    </select>
  </div>
);

const mapStateToProps = (state)=> {
  return {
    expenses:selectExpenses(state.expenses,state.filters),
    filters:state.filters
  };
};

export default connect(mapStateToProps)(Dropdown);
