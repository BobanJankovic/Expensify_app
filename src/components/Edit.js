import React from 'react';
import { connect } from 'react-redux';
import AddForm from './AddExpense';

import { startEditExpense,addExpense, removeExpense, startRemoveExpense } from '../actions/expenses';


const Edit = (props) => {
   console.log(props);
  return (
    <div>
    
      
      <AddForm
      //  This is edit od {props.expense.id}{props.expense.id} zbog linije koda 12 mi se vrednosti pojave u poljima koje treba da izmenim
      //props.expense pristupio sam iz stora i poslao je preko propsa
      expenseedit={props.expense}
       onSubmit={(expensed) => {
        
         props.dispatch(startEditExpense(props.expense.id, expensed));
          //props.dispatch(addExpense(expense));
          props.history.push('/');
         
        }}
         />
          <button  onClick={()=>{
            props.dispatch(startRemoveExpense({ id: props.expense.id}))
            props.history.push('/');
            }}>remove</button>
    </div>
  );
};


const mapStateToProps = (state, props) => {
  console.log(props);
  return {
    
    expense: state.expenses.find(expense => expense.id === props.match.params.id )
  };
};

export default connect(mapStateToProps)(Edit);
   
   /*import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(startEditExpense(props.expense.id, expense));
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }));
        props.history.push('/');
      }}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditExpensePage); */