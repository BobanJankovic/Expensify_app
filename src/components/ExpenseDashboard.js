import React from 'react';

import ExpensePage from '../components/ExpensePage';

export const ExpenseDashboard = () => (
    <div>
 
    <ExpensePage />

    </div>
);