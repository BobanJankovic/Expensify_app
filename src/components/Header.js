import React from 'react';
import { NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => (
  <header className="head">
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="selected" activeStyle={{fontWeight: 'bold',color: 'blue'}} exact={true}>ExpensePage</NavLink>
    <NavLink to="/dashboard" activeClassName="selected" activeStyle={{fontWeight: 'bold',color: 'pink'}} exact={true}>Dashboard Page</NavLink>
    <NavLink to="/createexpense" activeClassName="selected" activeStyle={{fontWeight: 'bold',color: 'green'}}  >Create Expense</NavLink>
    <NavLink to="/about" activeClassName="selected" activeStyle={{fontWeight: 'bold',color: 'gray'}}>About</NavLink>
    <NavLink to="/edit/:id" activeClassName="selected" activeStyle={{fontWeight: 'bold',color: 'green'}}>Edit</NavLink>
    <button className="logoutButton" onClick={ props.startLogout }>Log out</button>
  </header>
);


const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect( undefined, mapDispatchToProps )(Header);
