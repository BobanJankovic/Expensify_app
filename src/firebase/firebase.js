import * as firebase from 'firebase';


const config = {
  apiKey: "AIzaSyD2R8ZgGPEvaccZmgWz12uqP1EFRSlwOP4",
  authDomain: "expensify-925c6.firebaseapp.com",
  databaseURL: "https://expensify-925c6.firebaseio.com",
  projectId: "expensify-925c6",
  storageBucket: "expensify-925c6.appspot.com",
  messagingSenderId: "611814291707"
};

  firebase.initializeApp(config);

  const database = firebase.database();

   export { firebase, database as default };


//To read data at a path and listen for changes, use the on()

database.ref('expenses').on('value', function(snapShot) {
  const expenses=[];
  snapShot.forEach((childSnapshot)=>{
    expenses.push({
      id: childSnapshot.key,
      ...childSnapshot.val()
    })
  })
  //console.log(expenses);
});

// child_removed
database.ref('expenses').on('child_removed', (snapshot) => {
  //console.log(snapshot.key, snapshot.val());
});

// child_changed
database.ref('expenses').on('child_changed', (snapshot) => {
  //console.log(snapshot.key, snapshot.val());
});

// child_added
database.ref('expenses').on('child_added', (snapshot) => {
  //console.log(snapshot.key, snapshot.val());
});




/*
database.ref('expenses').push({
  description: "tv-waeeesh",
  note: "Wasdddhing tv-s",
  amount: 3704440,
  createdAt: 25488000
})



------------------------------------------------------------------------------------
Dec 2nd, 2018
1
firebase.js:50 -LSk7ZrmTDmBgx6Ok8P8 {amount: 3704440, createdAt: 25488000, description: "tv-waeeesh", note: "Wasdddhing tv-s"}

2
firebase.js:50 -LSjxAUTKScdweFvJK-j {amount: 3500, createdAt: 254588000, description: "Car-wash", note: "Washing cars"}
firebase.js:50 -LSjxAUsFxR97p4wxOrG {amount: 1500, createdAt: 2228000, description: "Bike-wash", note: "Washing bikes"}
firebase.js:50 -LSjxAUuQX2_YBRjL0dH {amount: 7500, createdAt: 774588000, description: "Truck-wash", note: "Washing truckss"}
3
firebase.js:35 (4) [{…}, {…}, {…}, {…}]

------------------------------------------------------------------------------------
database.ref('expenses').push({
  description: "Tracktor-wash",
  note: "Washing tractors",
  amount: 3700,
  createdAt: 2588000
})
database.ref('expenses').push({
  description: "tv-wash",
  note: "Washing tv-s",
  amount: 3700,
  createdAt: 2588000
})



database.ref('expenses').once('value').then((snapShot)=> {
  const expenses=[];
  snapShot.forEach((childSnapshot)=>{
    expenses.push({
      id: childSnapshot.key,
      ...childSnapshot.val()
    })
  })
  console.log(expenses);
})






database.ref('expenses').push({
  description: "Car-wash",
  note: "Washing cars",
  amount: 3500,
  createdAt: 254588000
})

database.ref('expenses').push({
  description: "Bike-wash",
  note: "Washing bikes",
  amount: 1500,
  createdAt: 2228000
})


database.ref('expenses').push({
  description: "Truck-wash",
  note: "Washing truckss",
  amount: 7500,
  createdAt: 774588000
})












database.ref('expenses/-LSemev6HBQHIVF5OXXS').once('value').then((snapshot) => {
      const val = snapshot.val(); 
      console.log(val);
    })
    



let expense = [];
var ref = firebase.database().ref("expense");
ref.once("value")
  .then(function(snapshot) {
   snapshot.forEach(function(childSnapshot) {
     expense.push({
         id:childSnapshot.key,
         ...childSnapshot.val()
        }
      )
   })
     console.log(expense);
    
  });


var query = firebase.database().ref("users").orderByKey();
query.once("value")
  .then(function(snapshot) {
    snapshot.forEach(function(childSnapshot) {
      // key will be "ada" the first time and "alan" the second time
      var key = childSnapshot.key;
      // childData will be the actual contents of the child
      var childData = childSnapshot.val();
  });
});

*/ 


 // database.ref('age').set(27);
  //database.ref('attributes').set({
    //wiight:130,
    //height:200
  //});
//let vrednost = [];
/*

database.ref().on('value', (snapshot) => {
vrednost.push(snapshot.val());
});
let obj = vrednost[0];
console.log(obj.surname);

  database.ref().set({
    name: 'Boban Jankovic',
    surname:'Jnakovic',
    age:25
  }).then(()=>{
    console.log('passed');
  }).catch((e)=>{
    console.log('notpassed',e);
  })
  

*/ 