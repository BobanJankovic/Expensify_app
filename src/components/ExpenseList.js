import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from "../components/ExpenseListItem";
import ExpenseListFilter from "../components/ExpenseListFilters";
import Dropdown from "../components/Dropdown";


//connect redux component to the redux store
const ExpenseList= (props)=> (
<div>
  <h1>Boban Jankovic je car</h1>
  <Dropdown />
  <ExpenseListItem />
  <ExpenseListFilter />

</div>
);

const mapStateToProps = (state)=> {
  return {
    expenses:state.expenses,
    filters:state.filters
  };
};
export default connect(mapStateToProps)(ExpenseList);

