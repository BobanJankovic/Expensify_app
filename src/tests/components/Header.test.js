import React from "react";
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {Header} from '../../components/Header';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import firebase from "../../firebase/firebase";

Enzyme.configure({ adapter: new Adapter() });

test('should render Header correctly', () => {
const wrapper = shallow(<Header startLogout={ () => {} } />);
expect(toJSON(wrapper)).toMatchSnapshot();
//expect(wrapper.find('h1').text()).toBe('Expensify');
//expect(renderer.getRenderOutput()).toMatchSnapshot();
//console.log(renderer.getRenderOutput());

});

// using spies

// should call startLogout on button click

// LoginPage test file -> Should call startLogin on button click
/*
test('should call onSubmit prop for valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<ExpenseForm expense={expenses[0]} onSubmit={onSubmitSpy} />);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error')).toBe('');
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    description: expenses[0].description,
    amount: expenses[0].amount,
    note: expenses[0].note,
    createdAt: expenses[0].createdAt
  });
});
*/ 
test('should call startLogout on button click', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<Header startLogout={ onSubmitSpy }  />);
  
 wrapper.find('button').simulate('click');
  expect(onSubmitSpy).toHaveBeenCalled();
});

