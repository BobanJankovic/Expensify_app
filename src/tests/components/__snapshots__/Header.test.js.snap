// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="head"
>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "blue",
        "fontWeight": "bold",
      }
    }
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    ExpensePage
  </NavLink>
  <NavLink
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "pink",
        "fontWeight": "bold",
      }
    }
    ariaCurrent="true"
    exact={true}
    to="/dashboard"
  >
    Dashboard Page
  </NavLink>
  <NavLink
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "green",
        "fontWeight": "bold",
      }
    }
    ariaCurrent="true"
    to="/createexpense"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "gray",
        "fontWeight": "bold",
      }
    }
    ariaCurrent="true"
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "green",
        "fontWeight": "bold",
      }
    }
    ariaCurrent="true"
    to="/edit/:id"
  >
    Edit
  </NavLink>
  <button
    onClick={[Function]}
  >
    Log out
  </button>
</header>
`;
